fusillade/api/__init__.py

```diff
@@ -0,0 +1,204 @@
+            if route.endswith("/"):
+                self.trailing_slash_routes.append(route.rstrip("/"))
+            routes[route.rstrip("/")] += rule.methods
+        for route, methods in routes.items():
+            self.route(route, methods=list(set(methods) - {"OPTIONS"}), cors=True)(self.dispatch)
+
+    def create_connexion_app(self):
+        app = FlaskApp('fusillade')
+        # The Flask/Connection app's logger has its own multi-line formatter and configuration. Rather than suppressing
+        # it we let it do its thing, give it a special name and only enable it if Fusillade_DEBUG > 1. Most of the Fusillade web
+        # app's logging is done through the FusilladeChaliceApp.app logger not the Flask app's logger.
+        app.app.logger_name = 'fus.api'
+        debug = Config.debug_level() > 0
+        app.app.debug = debug
+        app.app.logger.info('Flask debug is %s.', 'enabled' if debug else 'disabled')
+        
+        validator_map = {
+            'body': FusilladeRequestBodyValidator,
+            'parameter': FusilladeParameterValidator,
+        }
+
+        resolver = RestyResolver("fusillade.api", collection_endpoint_name="list")
+        app.add_api(self.swagger_spec_path,
+                    validator_map=validator_map,
+                    resolver=resolver,
+                    validate_responses=True,
+                    arguments=os.environ,
+                    options={"swagger_path": self.swagger_spec_path})
+        return app
+
+    def dispatch(self, *args, **kwargs):
+        """
+        This is the main entry point into the connexion application.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        cr = self.current_request
+        context = cr.context
+        uri_params = cr.uri_params or {}
+        method = cr.method
+        query_params = cr.query_params
+        path = context["resourcePath"].format(**uri_params)
+        if context["resourcePath"] in self.trailing_slash_routes:
+            if context["path"].endswith("/"):
+                path += "/"
+            else:
+                return chalice.Response(status_code=requests.codes.found, headers={"Location": path + "/"}, body="")
+        req_body = cr.raw_body if cr._body is not None else None
+        base_url = "https://{}".format(cr.headers["host"]) if cr.headers.get("host") else os.environ["API_DOMAIN_NAME"]
+        # TODO figure out of host should be os.environ["API_DOMAIN_NAME"]
+
+        self.log.info(
+            """[request] "%s %s" %s %s "%s" %s""",
+            method,
+            path,
+            context['identity']['sourceIp'],
+            cr.headers.get('content-length', '-'),
+            cr.headers.get('user-agent'),
+            str(query_params) if query_params is not None else '',
+        )
+        with self.connexion_request_context(path=path,
+                                                 base_url=base_url,
```

Comment:
Indent.

Reply:
the linter was con configured to recurse through the directories. fixed